src/controllers/hooks/router-store/index.tsx

```diff
@@ -18,3 +26,32 @@ export const useRouterActions = (): RouterActionsType => {
 
   return allActions;
 };
+
+export const useLocation = (): Location => {
+  const [routerState] = useRouterStore();
+
+  return routerState.location;
+};
+
+export const useHistory = (): BrowserHistory => {
+  const [routerState] = useRouterStore();
+
+  return routerState.history;
+};
+
+export const useParams = (): MatchParams => {
+  const [routerState] = useRouterStore();
+
+  return routerState.match.params;
+};
+
+export const useRouteMatch = (route?: string | Route): Match => {
```

Comment:
I'm not entirely sure we should expose this. Do we have an actual use case?

Reply:
What is the difference here between this and `useRouter`? Can't we get the current match from there?

Reply:
there's no difference really, this is more of a compatibility thing. I'll remove this one and `useHistory` as we don't really need to support them.